apps/wallet-mobile/src/components/QRCodeScanner/QRCodeScanner.tsx

```diff
@@ -30,15 +51,222 @@ export const QRCodeScanner = ({onRead}: {onRead: ({data}: {data: string}) => Pro
   }
 
   return (
-    // expo-barcode-scanner issue in android https://github.com/expo/expo/issues/5212
-    // so expo-camera is used
-    <Camera
-      style={StyleSheet.absoluteFill}
-      ratio="16:9"
-      barCodeScannerSettings={{
-        barCodeTypes: [BarCodeScanner.Constants.BarCodeType.qr],
-      }}
-      onBarCodeScanned={handleBarCodeScanned}
-    />
+    <>
+      {/*
+       * expo-barcode-scanner issue in android https://github.com/expo/expo/issues/5212
+       * so expo-camera is used
+       */}
+      <Camera
+        style={[StyleSheet.absoluteFill, styles.container]}
+        ratio="16:9"
+        barCodeScannerSettings={{
+          barCodeTypes: [BarCodeScanner.Constants.BarCodeType.qr],
+        }}
+        onBarCodeScanned={handleBarCodeScanned}
+      >
+        {maskEnabled && <Mask maskText={maskText} />}
+      </Camera>
+    </>
   )
 }
+
+const Mask = ({maskText}: {maskText: string}) => (
+  <>
+    <View style={styles.layerTop} />
```

Comment:
lf to leverage styled components, can we move for now to `<TopLayer> <CenteredLayer> <CameraOpening> <BottomLayer>` something like that, so many views hard to mentally visualize the ui representation